### PR TITLE
mailer - add ms graph api delivery for email

### PR DESCRIPTION
**Pull request by kentnsw**
This PR is to add the ability of sending email via MS Graph API.

_https://learn.microsoft.com/en-us/graph/overview
Microsoft Graph is the gateway to data and intelligence in Microsoft 365. It provides a unified programmability model that you can use to access the tremendous amount of data in Microsoft 365, Windows, and Enterprise Mobility + Security._

note: sorry for the spam due to failed those checks

### CODE REVIEW

File: tools/c7n_mailer/c7n_mailer/email_delivery.py [hunk 1]
```diff
@@ -26,7 +27,6 @@ def __init__(self, config, session, logger):
         if self.provider == Providers.AWS:
             self.aws_ses = self.get_ses_session()
         self.ldap_lookup = self.get_ldap_connection()
-        self.provider = get_provider(self.config)
```
Comment (ajkerrigan) [inline tools/c7n_mailer/c7n_mailer/email_delivery.py:29]: Ah good catch :+1:

**Review by ajkerrigan — APPROVED** [review]
Neat, thanks for adding this support :+1: . Looks good to me.

I'm wondering if we have enough branching paths in mailer config now that it's worth doing a bit more validation of the possible parameter combinations :thinking: . That'd be separate from this PR though.

**Comment by kentnsw** [comment]
> it's worth doing a bit more validation of the possible parameter combinations

Totally agree.

**Comment by kapilt** [comment]
Not sure about this one, graph is an administrative api. Versus primarily
communication, like azure communications, ie graph it’s used to read
inboxes as well. It’s not least privilege for sending

**Comment by kentnsw** [comment]
> Not sure about this one, graph is an administrative api...

Hi @kapilt , could you please elaborate it a bit? eg, 
- is Graph API not suggested to be used to send mail?
- does the way we obtain the token have a too broad scope? 
- and how could we improve it?

My case is that I am told MS Graph is the default (and only) way to send mail in my org; so I don't have much choice. The service account permission is limited send mail, I was told.

**Comment by kapilt** [comment]
I was viewing graph api as more of an admin api, I've heard its usage in orgs is generally restricted due to its overarching scope across all 365/mobile/azure data, albeit that can likely be limited via permissioning model. azure communications services is a more targeted service for delivering communications for app to person. 

https://azure.microsoft.com/en-us/products/communication-services

https://medium.com/medialesson/how-to-send-emails-at-scale-in-net-with-the-azure-communication-service-14565d84147f

**Pull request merged**